modules/develop/pages/data-transforms/run-transforms.adoc

```diff
@@ -0,0 +1,254 @@
+func identityTransform(e redpanda.WriteEvent) ([]redpanda.Record, error) {
+	return []redpanda.Record{e.Record()}, nil
+}
+```
+--
+Transcoder transform::
++
+--
+
+```go
+package main
+
+import (
+	"bytes"
+	"encoding/csv"
+	"encoding/json"
+	"errors"
+	"io"
+	"strconv"
+
+	"github.com/redpanda-data/redpanda/src/go/transform-sdk"
+)
+
+// This example shows a transform that converts CSV inputs into JSON outputs.
+func main() {
+	redpanda.OnRecordWritten(csvToJsonTransform)
+}
+
+type Foo struct {
+	A string `json:"a"`
+	B int    `json:"b"`
+}
+
+func csvToJsonTransform(e redpanda.WriteEvent) ([]redpanda.Record, error) {
+	// The input data is a CSV (without a header row) that is the structure of:
+	// key, a, b
+	reader := csv.NewReader(bytes.NewReader(e.Record().Value))
+	// Improve performance by reusing the result slice.
+	reader.ReuseRecord = true
+	output := []redpanda.Record{}
+	for {
+		row, err := reader.Read()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return nil, err
+		}
+		if len(row) != 3 {
+			return nil, errors.New("unexpected number of rows")
+		}
+		// Convert the last column into an int
+		b, err := strconv.Atoi(row[2])
+		if err != nil {
+			return nil, err
+		}
+		// Marshal our JSON value
+		f := Foo{
+			A: row[1],
+			B: b,
+		}
+		v, err := json.Marshal(&f)
+		if err != nil {
+			return nil, err
+		}
+		// Add our output record using the first column as the key.
+		output = append(output, redpanda.Record{
+			Key:   []byte(row[0]),
+			Value: v,
+		})
+
+	}
+	return output, nil
+}
+```
+
+--
+Validation filter transform::
++
+--
+```go
+import (
+	"encoding/json"
+
+	"github.com/redpanda-data/redpanda/src/go/transform-sdk"
+)
+
+// This example shows a filter that outputs only valid JSON into the
+// output topic.
+func main() {
+	redpanda.OnRecordWritten(filterValidJson)
+}
+
+func filterValidJson(e redpanda.WriteEvent) ([]redpanda.Record, error) {
+	v := []redpanda.Record{}
+	if json.Valid(e.Record().Value) {
+		v = append(v, e.Record())
+	}
+	return v, nil
+}
+```
+
+--
+====
+
+== Build and deploy the transform
+
+. Build the transform into a Wasm module with metadata: 
++
+```bash
+rpk transform build
+```
+
+. Deploy the Wasm module to your cluster. For example:
++
+```bash
+rpk transform deploy --input-topic=demo-1 --output-topic=demo-2
+```
+
+. Validate that your transform is running. For example:
+.. Produce a few records to the `demo-1` topic.
++
+```bash
+echo "foo\nbar" | rpk topic produce demo-1
+```
+.. Consume from the `demo-2` topic.
++
+```bash
+rpk topic consume demo-2
+```
++
+[,json,role="no-copy"]
+----
+{
+  "topic": "demo-2",
+  "value": "foo",
+  "timestamp": 1687545891433,
+  "partition": 0,
+  "offset": 0
+}
+
+  "topic": "demo-2",
+  "value": "bar",
+  "timestamp": 1687545892434,
+  "partition": 0,
+  "offset": 1
+}
+----
+
+NOTE: You can see `stdout` and `stderr` for your function in the broker's logs. 
```

Comment:
@rockwotj  besides user-supplied stdout/stderr lines, we should also quote a couple of transform lifecycle log lines here so people know exactly what to look for. Or maybe even show both and point out which is which.    People really are going to need a hand debugging

Reply:
I still need to do a full overview/audit of logging. Let's follow up on this in the next PR, but agree, stuff is still landing on dev for this.